### DISCUSSION TITLE
How to handle a deferred column_property?

### DISCUSSION THREAD

**Discussion by michaelgmiller1** [pydantic/pydantic#6662] [Question]
Our app has several models that use `column_property` to execute expensive queries. We set them to `deferred=False`, so we can manually control when they are queried:

```
class ExampleModel(BaseModel):
    class Config:
        orm_mode = True

   # other columns defined here....
    my_expensive_column: str
```

```
class ExampleTable(BaseSQLAlchemy):
  # other columns defined here
  my_expensive_column = column_property(..., deferred=True)
```

Then we do this to query:

```
query = db.query(ExampleTable)
# Optionally undefer depending on code path
if should_undefer:
  query = query.options(undefer('my_expensive_column'))
result = query.one()
model_result = self.model.from_orm(result) if result else None
```

When we run this code path, we find that `my_expensive_column` is queried. `from_orm` calls `pydantic.main.validate_model` which calls `pydantic.utils.GetterDict.get`, which causes SQLAlchemy to fetch the column.

We'd like the column to remain un-fetched, unless explicitly un-deferred. Is there a good way to do this?

**Comment by tkrause** [comment]
Looking to do the same anyone have a solution for this?

**Comment by jordantshaw** [comment]
Also facing this issue.  Anyway to disable the implicit lazy loading that is happening on specific fields?

@samuelcolvin @dmontagu
